common4j/src/main/com/microsoft/identity/common/java/telemetry/events/ApiStartEvent.java

```diff
@@ -0,0 +1,252 @@
+
+            final SilentTokenCommandParameters silentParameters = (SilentTokenCommandParameters) parameters;
+
+            if (silentParameters.getAccount() != null) {
+                put(Key.USER_ID, silentParameters.getAccount().getHomeAccountId()); //Pii
+            }
+            put(
+                    Key.IS_FORCE_REFRESH,
+                    String.valueOf(silentParameters.isForceRefresh())
+            );
+        }
+
+        if (parameters instanceof BrokerInteractiveTokenCommandParameters) {
+            //TODO when integrate the telemetry with broker.
+        }
+
+        if (parameters instanceof BrokerSilentTokenCommandParameters) {
+            final BrokerSilentTokenCommandParameters silentParameters = (BrokerSilentTokenCommandParameters) parameters;
+        }
+
+        return this;
+    }
+
+    public ApiStartEvent authority(@NonNull final String authority) {
+        put(Key.AUTHORITY, sanitizeUrlForTelemetry(authority));
+        return this;
+    }
+
+    public ApiStartEvent putAuthorityType(@NonNull final String authorityType) {
+        put(Key.AUTHORITY_TYPE, authorityType);
+        return this;
+    }
+
+    public ApiStartEvent putApiId(@NonNull final String apiId) {
+        put(Key.API_ID, apiId);
+        return this;
+    }
+
+    public ApiStartEvent putValidationStatus(@NonNull final String validationStatus) {
+        put(Key.AUTHORITY_VALIDATION_STATUS, validationStatus);
+        return this;
+    }
+
+    public ApiStartEvent putLoginHint(@NonNull final String loginHint) {
+        final String methodTag = TAG + ":putLoginHint";
+        try {
+            put(Key.LOGIN_HINT, StringUtil.createHash(loginHint));
+        } catch (final NoSuchAlgorithmException exception) {
+            Logger.warn(methodTag, exception.getMessage());
+        }
+
+        return this;
+    }
+
+    public ApiStartEvent putExtendedExpiresOnSetting(@NonNull final String extendedExpiresOnSetting) {
+        put(Key.EXTENDED_EXPIRES_ON_SETTING, extendedExpiresOnSetting);
+        return this;
+    }
+
+    /**
+     * Convenience method to sanitize the url for telemetry.
+     *
+     * @param url the {@link URL} to sanitize.
+     * @return the sanitized URL.
+     */
+    private static String sanitizeUrlForTelemetry(@NonNull final String url) {
+        URL urlToSanitize = null;
+        try {
+            urlToSanitize = new URL(url);
+        } catch (MalformedURLException e1) {
+            Logger.errorPII(
+                    TAG,
+                    "Url is invalid",
+                    e1
+            );
+        }
+
+        return urlToSanitize == null ? null : sanitizeUrlForTelemetry(urlToSanitize);
+    }
+
+    /**
+     * Sanitizes {@link URL} of tenant identifiers. B2C authorities are treated as null.
+     *
+     * @param url the URL to sanitize.
+     * @return the sanitized URL.
+     */
+    private static String sanitizeUrlForTelemetry(@NonNull final URL url) {
+        if (url == null) {
```

Comment:
The param is already marked NonNull here

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1757/commits/49bfef79d16c8e9616933edb33160b6bfe0695e5